src/application/look-up-wallet-address/look-up-wallet-address.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from 'react';
+import { useKey } from 'react-use';
+import { useForm } from 'react-hook-form';
+
+import { useCommandQuery } from 'shared/messaging';
+import { Closable, IconButton, Spinner } from 'shared/ui';
+import { useExtensionSettings } from 'shared/extension';
+
+import { GetResolvedAddressCommand, GetTwitterIdsCommand } from './commands';
+import { AddressList } from './address-list';
+
+export const LookUpWalletAddress = () => {
+  const [username, setUsername] = useState('');
+  const [isVisible, setIsVisible] = useState(false);
+
+  const { extensionSettings } = useExtensionSettings();
+
+  const isEnabled =
+    extensionSettings['entire-extension-enabled'] &&
+    extensionSettings['wallet-lookup-enabled'];
+
+  const twitterIdsQuery = useCommandQuery({
+    command: new GetTwitterIdsCommand({
+      username,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+  });
+
+  const addressesQuery = useCommandQuery({
+    command: new GetResolvedAddressCommand({
+      identifier: username,
+      twitterId: twitterIdsQuery.data?.twitterIDs
+        ? Object.values(twitterIdsQuery.data?.twitterIDs)[0]
+        : undefined,
+    }),
+    staleTime: Number.POSITIVE_INFINITY,
+    enabled: username.length >= 3,
+  });
+
+  const form = useForm();
+
+  const userIdentifier = form.watch('user_identifier');
+
+  const onSubmit = () => {
+    const values = form.getValues();
+    setUsername(values.user_identifier);
+  };
```

Comment:
with react-hook-form you don't need to manually call getValues, your submit handler will receive data from form, you should also add correct type:
```tsx
type LookupFormValues = {
  identifier: string;
}

const EMPTY_LOOKUP_FORM_VALUES: LookupFormValues = {
  identifier: '',
};

const form = useForm<LookupFormValues>({ defaultValues: EMPTY_LOOKUP_FORM_VALUES });

const onSubmit: SubmitHandler<LookupFormValues> = (data) => {
  setUsername(data);
}
```